extension/observer/cfgardenobserver/config.go

```diff
@@ -4,15 +4,116 @@
 package cfgardenobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver"
 
 import (
+	"errors"
+	"fmt"
 	"time"
 )
 
 // Config defines configuration for CF Garden observer.
 type Config struct {
-	// The URL of the CF Garden api. Default is "unix:///var/vcap/data/garden/garden.sock"
-	Endpoint string `mapstructure:"endpoint"`
+	// CloudFoundry API Configuration
+	CloudFoundry CfConfig `mapstructure:"cloud_foundry"`
+
+	// Garden API Configuration
+	Garden GardenConfig `mapstructure:"garden"`
 
 	// RefreshInterval determines the frequency at which the observer
 	// needs to poll for collecting information about new processes.
+	// Default: "1m"
 	RefreshInterval time.Duration `mapstructure:"refresh_interval"`
+
+	// The time to wait before resyncing app information on cached containers
+	// using the CloudFoundry API. Example: cache_sync_interval: "20m"
+	// Default: "5m"
+	CacheSyncInterval time.Duration `mapstructure:"cache_sync_interval"`
+
+	// Determines whether or not Application labels get added to the Endpoint labels.
+	// This requires cloud_foundry to be configured, such that API calls can be made
+	// Default: false
+	IncludeAppLabels bool `mapstructure:"include_app_labels"`
+}
+
+// Validate overrides the embedded noop validation so that load config can trigger
+// our own validation logic.
+func (config *Config) Validate() error {
+	if !config.IncludeAppLabels {
+		return nil
+	}
+
+	c := config.CloudFoundry
+	if c.Endpoint == "" {
+		return errors.New("config.Endpoint must be specified when include_app_labels is set to true")
+	}
+	if c.AuthType == "" {
+		return errors.New("config.AuthType must be specified when include_app_labels is set to true")
+	}
+
+	switch c.AuthType {
+	case AuthTypeUserPass:
+		if c.Username == "" {
+			return fieldError(AuthTypeUserPass, "username")
+		}
+		if c.Password == "" {
+			return fieldError(AuthTypeUserPass, "password")
+		}
+	case AuthTypeClientCredentials:
+		if c.ClientID == "" {
+			return fieldError(AuthTypeClientCredentials, "client_id")
+		}
+		if c.ClientSecret == "" {
+			return fieldError(AuthTypeClientCredentials, "client_secret")
+		}
+	case AuthTypeToken:
+		if c.AccessToken == "" {
+			return fieldError(AuthTypeToken, "access_token")
+		}
+		if c.RefreshToken == "" {
+			return fieldError(AuthTypeToken, "refresh_token")
+		}
+	default:
+		return fmt.Errorf("unknown auth_type: %s", c.AuthType)
```

Comment:
```suggestion
		return fmt.Errorf("configuration option `auth_type` must be set to one of the following values: [user_pass, client_credentials, token]. Specified value: %s", c.AuthType)
```
It would be good to show that it's required, what values are valid, and what the user specified here.